packages/scanner/src/ruleChecker.ts

```diff
@@ -145,19 +180,87 @@ export default class RuleChecker {
         additionalEvents || []
       );
 
+      let scopeModifiedDate: Date | undefined;
+      {
+        const scopeNavigator = new EventNavigator(scope);
+        const scopeFiles = new Set<string>();
+        const collectScope = (event: Event) => {
+          if (!event.codeObject.location) return;
+
+          const filePath = locationToFilePath(event.codeObject.location);
+          if (!filePath) return;
+
+          scopeFiles.add(filePath);
+        };
+
+        collectScope(scope);
+        for (const descendant of scopeNavigator.descendants()) {
+          const { event } = descendant;
+          collectScope(event);
+        }
+
+        const localScopeFiles = [...scopeFiles].filter(
+          (filePath) => (assert(filePath), !isAbsolute(filePath))
+        ) as string[];
+        scopeModifiedDate = (
+          await Promise.all(
+            localScopeFiles.map(async (filePath) => {
+              const resolvedPath = await resolvePath(filePath);
+              if (!resolvedPath) return;
+
+              return lastGitOrFSModifiedDate(resolvedPath);
+            })
+          )
```

Comment:
This should probably loop instead of using `Promise.all` to avoid potentially spawning dozens of `git` processes simultaneously.